pkg/pgmodel/end_to_end_tests/query_integration_test.go

```diff
@@ -1124,3 +1130,82 @@ func generateLargeTimeseries() []prompb.TimeSeries {
 
 	return metrics
 }
+
+func TestPushdown(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	testCases := []struct {
+		name    string
+		query   string
+		startMs int64
+		endMs   int64
+		stepMs  int64
+		res     promql.Result
+	}{
+		{
+			name:    "Simple metric name matcher",
+			query:   `delta(metric_1{instance="1"}[5m])`,
+			startMs: startTime + 300*1000,
+			endMs:   startTime + 330*1000,
+			stepMs:  30 * 1000,
+			res: promql.Result{
+				Value: promql.Matrix{promql.Series{
+					Points: []promql.Point{{V: 20, T: startTime + 300000}, {V: 20, T: startTime + 330000}},
+					Metric: labels.FromStrings("foo", "bar", "instance", "1")},
+				},
+			},
+		},
+		{
+			name:  "Simple metric name matcher",
+			query: `delta(metric_1{instance="1"}[5m])`,
+			endMs: startTime + 300*1000,
+			res: promql.Result{
+				Value: promql.Vector{promql.Sample{
+					Point:  promql.Point{V: 20, T: startTime + 300*1000},
+					Metric: labels.FromStrings("foo", "bar", "instance", "1")},
+				},
+			},
+		},
+	}
+
+	withDB(t, *testDatabase, func(db *pgxpool.Pool, t testing.TB) {
+		// Ingest test dataset.
+		ingestQueryTestDataset(db, t, generateLargeTimeseries())
+		// Getting a read-only connection to ensure read path is idempotent.
+		readOnly := testhelpers.GetReadOnlyConnection(t, *testDatabase)
+		defer readOnly.Close()
+
+		var tester *testing.T
+		var ok bool
+		if tester, ok = t.(*testing.T); !ok {
+			t.Fatalf("Cannot run test, not an instance of testing.T")
+			return
+		}
```

Comment:
is it really worth passing the `testing.TB` out of `withDB` if we're just going to have to do stuff like this? Realistically it's a bug if `withDB` ever swaps out the `testing.<T/B>` for anything else, so we can just capture the `testing.<T/B>` from the environment